### ISSUE TITLE
Equal jobs don't compare equal

### ISSUE THREAD

**Issue by pmrv** [pyiron/pyiron_base#59]
Currently `GenericJob` doesn't implement `__eq__` so two job instances of the same class, with the same name in the same project don't compare as equal
```python
(job1 == job2) == False
```
This is not strictly needed anywhere right now, but would be a nice quality of life improvement, since you can stick them InputLists soon.

**Comment by samwaseda** [comment]
Will it be also possible to show the differences?

**Comment by pmrv** [comment]
Not sure what you're asking for.  This would be more for a case where you get two instances of jobs from different sources (e.g. `pr.iter_jobs` and `pr.load`) and make sure they're the same. Right now you can do `job1.id == job2.id` just as well, but overloading might be convenient.

**Comment by samwaseda** [comment]
Oh I thought you'd compare the inputs, but you're just comparing the job names? That sounds like a niche problem.

**Comment by pmrv** [comment]
Yup, that's why nothing has been happening yet. :') 

So for your suggestion the use case would be that you create a new job and want to check whether another one runs the same calculation+parameters?

**Comment by samwaseda** [comment]
> Yup, that's why nothing has been happening yet. :')
> 
> So for your suggestion the use case would be that you create a new job and want to check whether another one runs the same calculation+parameters?

Yes, because it happens often that I can't remember whether I've already chosen a specific parameter set. I'm relatively sure that some people have faced the same issue, so it would be definitely helpful if there could be an input comparison. This I guess will also make it easier to implement a big data infrastructure at some point.

**Comment by samwaseda** [comment]
Solution: `job1 == job2` compares their job ID's if they exist, otherwise always `False`.
